kratos/geometries/geometry.h

```diff
@@ -690,7 +690,7 @@ class Geometry : public PointerVector<TPointType>
      * It computes the normal of the geometry, if possible
      * @return The normal of the geometry
      */
-    virtual array_1d<double, 3> Normal(CoordinatesArrayType& rPointLocalCoordinates)
+    virtual array_1d<double, 3> Normal(const CoordinatesArrayType& rPointLocalCoordinates)
```

Comment:
a hint: you already implemented the normal in all of the geometries. At this point why don't you make this "virtual pure"?

     virtual array_1d<double, 3> Normal(const CoordinatesArrayType& rPointLocalCoordinates) = 0

this way anyone implementing a new geometry will HAVE to implement it